services/visual_recognition/v3.js

```diff
@@ -62,301 +37,329 @@ module.exports = function (RED) {
-    }
+    return true;
   }
 
-  // This is the Watson Visual Recognition V3 Node
-  function WatsonVisualRecognitionV3Node (config) {
-    RED.nodes.createNode(this, config);
-    var node = this;
-    var message;
-    this.on('input', function (msg) 
-    {
-      // Check which single feature has been requested.
-      var feature = config['image-feature'];
-      node.status({});
-      // so there is at most 1 temp file at a time (did not found a better solution...)
-      temp.cleanup(); 
 
-      if (!msg.payload) {
-        this.status({fill:'red', shape:'ring', text:'missing payload'}); 
-        message = 'Missing property: msg.payload';
-        node.error(message, msg);
-        return;
+  function processResponse(err, body, feature, node, msg) {
+    if (err != null && body == null) {
+      node.status({fill:'red', shape:'ring', 
+        text:'call to watson visual recognition v3 service failed'}); 
+      msg.result = {};
+      if (err.code == null) {
+        msg.result['error']=err;
+      } else {
+        msg.result['error_code'] = err.code;
+        if (!err.error) {
+          msg.result['error'] = err.error;
+        }        
       }
-      if (feature !== 'retrieveClassifiersList' && feature !== 'retrieveClassifierDetails' 
-          && feature !== 'deleteClassifier')
-      {
-        if (typeof msg.payload === 'boolean' || typeof msg.payload === 'number') 
-        {
-          this.status({fill:'red', shape:'ring', text:'bad format payload'}); 
-          message = 'Bad format : msg.payload must be a URL string or a Node.js Buffer';
-          node.error(message, msg);
-          return;
-        }
+      node.error(err);
+      return;
+    } else if (err == null && body != null && body.images != null && 
+      body.images[0].error) {
+      node.status({fill:'red', shape:'ring', 
+                   text:'call to watson visual recognition v3 service failed'}); 
+      msg.result = {};
+      msg.result['error_id'] = body.images[0].error.error_id;
+      msg.result['error'] = body.images[0].error.description;
+      msg.payload = 'see msg.result.error';
+      node.send(msg); 
+    } else {
+      if (feature === 'deleteClassifier') {
+        msg.result = 'Successfully deleted classifier_id: ' + msg.params.classifier_id ;
+      } else {
+        msg.result = body;
       }
+      msg.payload = 'see msg.result'; // to remove any Buffer that could remains
+      node.send(msg); 
+      node.status({});
+    }
+  }
 
-      // If it is present the newly provided user entered key 
-      // takes precedence over the existing one. 
-      apikey = sAPIKey || this.credentials.apikey;
-      this.status({}); 
-
-      if (!apikey) {
-        this.status({fill:'red', shape:'ring', text:'missing credentials'});          
-        message ='Missing Watson Visual Recognition API service credentials'; 
-        node.error(message, msg);
-        return;
+  function prepareParamsCommon(params, node, msg, cb) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'prepareParamsCommon' has a complexity of 6.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2737057949/issues/source?bid=3225115&fileBranchId=3352493#l138)